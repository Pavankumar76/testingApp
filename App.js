import React, { useState } from 'react';
import { View, StyleSheet, Text } from 'react-native';
import AddList from './components/AddList';
import ItemsList from './components/ItemsList/Item';
import Error from './components/ErrorComponent/Error';

export default function App() {
  const [listInput, setListInput] = useState('');
  const [list, setList] = useState([]);
  const [error, setError] = useState(false);

  const handleInput = (text) => {
    setListInput(text);
  };

  const handleSetList = () => {
    if (!listInput) {
      setError(true);
      return;
    }

    const newInput = {
      id: Math.random() * 1000,
      value: listInput,
    };
    setList([...list, newInput]);
    setListInput('');
    setError(false);
  };

  const handleDelete = (id) => {
    setList(list.filter((item) => item.id !== id));
  };

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.headerText}>To-Do-Lists</Text>
      </View>
      <View style={styles.listArea}>
        <ItemsList data={list} onDelete={handleDelete} />
        <Error visible={error}>Please insert a valid text</Error>
      </View>
      <View style={styles.inputArea}>
        <View style={styles.inputHeaderView}>
          <Text style={styles.inputHeader}>Add Task</Text>
        </View>
        <AddList
          onPressAdd={handleSetList}
          value={listInput}
          onChange={handleInput}
        />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    marginTop: 10,
  },
  inputArea: {
    justifyContent: 'flex-end',
    marginBottom: 20,
  },
  listArea: {
    flex: 1,
    paddingTop: 10,
    alignItems: 'center',
    width: '80%',
    marginVertical: 15,
  },
  header: {
    marginTop: 40,
  },
  headerText: {
    alignItems: 'center',
    fontSize: 20,
    fontWeight: 'bold',
    color: 'blue',
  },
  inputHeader: {
    fontSize: 18,
    fontWeight: 'bold',
    textAlign: 'left',
    color: 'darkblue',
  },
  inputHeaderView: {
    alignItems: 'flex-start',
  },
});
