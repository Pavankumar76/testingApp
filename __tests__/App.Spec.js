import React from 'react';
import App from '../App';
import { render, fireEvent } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

it('should create item', () => {
  const { getByText, getByPlaceholderText } = render(<App />);

  const addItemButton = getByText('+');
  const textInput = getByPlaceholderText('Write something');
  const createdItemText = 'Learn ReactNative';
  fireEvent.changeText(textInput, createdItemText);
  fireEvent.press(addItemButton);
  const createdItem = getByText(createdItemText);

  expect(createdItem).not.toBeNull();
});

it('should create multiple item', () => {
  const { getByText, getByPlaceholderText } = render(<App />);

  const addItemButton = getByText('+');
  const textInput = getByPlaceholderText('Write something');

  const createdItemText = 'Learn ReactNative';
  const createdItemText_2 = 'Learning ReactNavigation';

  fireEvent.changeText(textInput, createdItemText);
  fireEvent.press(addItemButton);

  fireEvent.changeText(textInput, createdItemText_2);
  fireEvent.press(addItemButton);

  const firstcreatedItem = getByText(createdItemText);
  const secondcreatedItem = getByText(createdItemText_2);

  expect(firstcreatedItem).not.toBeNull();
  expect(secondcreatedItem).not.toBeNull();
});
it('should delete item', () => {
  const { getByText, getByPlaceholderText, queryByText } = render(<App />);

  const addItemButton = getByText('+');
  const textInput = getByPlaceholderText('Write something');

  const createdItemText = 'Learn ReactNative';

  fireEvent.changeText(textInput, createdItemText);
  fireEvent.press(addItemButton);

  const deleteItemButton = getByText('X');

  fireEvent.press(deleteItemButton);

  const deletedItem = queryByText(createdItemText);

  expect(deletedItem).toBeNull();
});
it('should display error message while creating item with out text', () => {
  const { getByText } = render(<App />);

  const addItemButton = getByText('+');

  fireEvent.press(addItemButton);

  const errorMessage = getByText('Please insert a valid text');

  expect(errorMessage).not.toBeNull();
});
it('should remove error after entering the text', () => {
  const { getByText, getByPlaceholderText, queryByText } = render(<App />);

  const addItemButton = getByText('+');

  fireEvent.press(addItemButton);

  const textInput = getByPlaceholderText('Write something');
  const createdItemText = 'Learn ReactNative';
  fireEvent.changeText(textInput, createdItemText);
  fireEvent.press(addItemButton);

  const errorMessage = queryByText('Please insert a valid text');

  expect(errorMessage).toBeNull();
});
it('App snapshot', () => {
  const snapshot = renderer.create(<App />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
