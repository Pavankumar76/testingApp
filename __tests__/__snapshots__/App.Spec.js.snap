// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "marginTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 40,
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "center",
          "color": "blue",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      To-Do-Lists
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginVertical": 15,
        "paddingTop": 10,
        "width": "80%",
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <RCTScrollView
        data={Array []}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
    <View />
  </View>
  <View
    style={
      Object {
        "justifyContent": "flex-end",
        "marginBottom": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "darkblue",
            "fontSize": 18,
            "fontWeight": "bold",
            "textAlign": "left",
          }
        }
      >
        Add Task
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Write something"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomColor": "black",
            "borderBottomWidth": 1,
            "width": 275,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
            "borderRadius": 40,
            "marginLeft": 15,
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 32,
            }
          }
        >
          +
        </Text>
      </View>
    </View>
  </View>
</View>
`;
